### PR TITLE
Prl 2235

### PR DESCRIPTION
**Pull request by praveenrach**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/document/DocumentManagementController.ts [hunk 1]
```diff
@@ -132,60 +158,19 @@ export class DocumentManagerController extends PostController<AnyObject> {
...
-      req.session.user = caseworkerUser;
-      const caseReference = req.session.userCase.id;
+      loggedInCitizen = req.session.user;
+      caseReference = req.session.userCase.id;
 
-      const client = new CosApiClient(caseworkerUser.accessToken, 'https://return-url');
-      const caseDataFromCos = await client.retrieveByCaseId(caseReference, caseworkerUser);
+      client = new CosApiClient(loggedInCitizen.accessToken, 'https://return-url');
+      const caseDataFromCos = await client.retrieveByCaseId(caseReference, loggedInCitizen);
       req.session.userCase = caseDataFromCos;
```
Comment (sahaSonali) [inline src/main/app/document/DocumentManagementController.ts:166]: Plz use object.assign to set usercase vale from cos

File: src/main/app/document/DocumentManagementController.ts [hunk 2]
```diff
@@ -303,22 +230,163 @@ export class DocumentManagerController extends PostController<AnyObject> {
...
+        respondent.value.response.citizenFlags = {
+          isAllegationOfHarmViewed: 'No',
+          isApplicationViewed: 'Yes',
+        };
+      } else {
+        respondent.value.response.citizenFlags = {
+          isAllegationOfHarmViewed: 'Yes',
+          isApplicationViewed: 'Yes',
+        };
+      }
```
Comment (sahaSonali) [inline src/main/app/document/DocumentManagementController.ts:355]: Plz update only one field which is required

File: src/main/steps/respondent/task-list/tasklist.ts [hunk 3]
```diff
@@ -84,29 +84,29 @@ export const generateRespondentTaskList = (sectionTitles, taskListItems, userCas
...
   const itemList: object[] = [];
   if (userCase?.caseTypeOfApplication === 'C100') {
     itemList.push(
       {
         id: 'check_the_application',
         text: taskListItems.check_the_application,
-        status: getMiamStatus(userCase),
-        href: URL.MIAM_START,
+        status: getFinalApplicationStatus(userCase, userIdamId),
+        href: URL.APPLICANT_CA_DA_REQUEST + '?updateCase=Yes',
```
Comment (sahaSonali) [inline src/main/steps/respondent/task-list/tasklist.ts:95]: Plz put it in a constant

Comment (praveenrach) [reply src/main/steps/respondent/task-list/tasklist.ts]: code changes done

File: src/main/app/document/DocumentManagementController.ts (hunk 2 above)
Comment (praveenrach) [reply src/main/app/document/DocumentManagementController.ts:352]: code changes done

File: src/main/app/document/DocumentManagementController.ts (hunk 1 above)
Comment (praveenrach) [reply src/main/app/document/DocumentManagementController.ts]: code changes done

**Comment by AlokDatta** [comment]
Closing this PR. As covered in 477

**Pull request closed without merging**